frontend/parent/iframe.ts

```diff
@@ -1,32 +1,55 @@
 import { containerClass, iframeClass } from "./util";
 
+const LOGIN_TOKEN_PARAM_NAME = "loginToken";
+
 export type IframeParams = {
     defaultHomeserver: string
     roomId: string,
 }
 
-export class Iframe {
-    private readonly _url: string;
-    private readonly iframe: HTMLIFrameElement;
+export class IframeUrl {
+    private readonly url: URL;
 
     constructor(hostRoot: string, params: IframeParams) {
-        this._url = this.makeUrl(hostRoot, params);
+        this.url = new URL("index.html?", hostRoot);
 
-        this.iframe = document.createElement("iframe");
-        this.iframe.src = this.url;
-        this.iframe.className = iframeClass();
+        for (let key in params) {
+            if (!!params[key]) {
+                this.url.searchParams.append(key, params[key]);
+            }
+        }
+
+        this.applyLoginToken();
+    }
+
+    public toString(): string {
+        return this.url.toString();
     }
 
-    public get url(): string {
-        return this._url;
+    public hasLoginToken(): boolean {
+        return this.url.searchParams.has(LOGIN_TOKEN_PARAM_NAME);
     }
 
-    public get element(): HTMLIFrameElement {
-        return this.iframe;
+    private applyLoginToken(): void {
+        const paramName = LOGIN_TOKEN_PARAM_NAME;
+        const queryParams = new URLSearchParams(window.location.search);
+
+        if (queryParams.has(paramName)) {
+            this.url.searchParams.append(
+                paramName,
+                queryParams.get(paramName) ?? ""
+            );
+        }
     }
+}
 
-    public get visible(): boolean {
-        return this.iframe.style.display !== "none";
+export class Iframe {
```

Comment:
Is it ok to have two classes in a file? (Maybe I am just spoiled by WordPress coding standards)

Reply:
I actually considered moving `IframeUrl` to its own file, but the file would be called `iframe-url.ts` which didn't feel right :D That said, it is somewhat common, at least in the hydrogen codebase, to have related classes in the same file.

I think we should rename all files to be `CamelCase.ts`, and then extract each class to its own file.